tsdb/store.go

```diff
@@ -155,6 +155,12 @@ func (s *Store) loadShards() error {
 				return err
 			}
 			for _, sh := range shards {
+				// Check to see if there are any `.tmp` directories that were left over from failed shard snapshots
+				if sh.IsDir() && strings.HasSuffix(sh.Name(), ".tmp") {
```

Comment:
To be clear, you are suggesting that we create a `Engine.cleanup` method and call it from this spot correct?